### PR TITLE
Add Project Evaluate view for compactness

### LINKED ISSUE #607: Evaluate mode compactness
image

AC:
 - Update evaluate mode summary sidebar to display average compactness value
 - Show compactness layer, map legend, and sidebar legend when switching to compactness detail

### PR DESCRIPTION
**Pull request by kevinearldenny**
## Overview

Adds project evaluate mode view for compactness.

- Computes the average compactness and displays it in `ProjectEvaluateSummary`
- Display list of districts and their compactness score in metric detail view
- Display chloropleth map in `Map`
- Add legend for chloropleth map

### Checklist

- [x] Description of PR is in an appropriate section of `CHANGELOG.md` and grouped with similar changes, if possible

### Demo

image

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

- `./scripts/server`
- Open a project and click evaluate
- Click 'Compactness'
- Expect: Compactness view is rendered as pictured

Closes #607

### CODE REVIEW

File: src/client/components/evaluate/detail/Compactness.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    while (i < chloroplethSteps.length) {
+      const r = chloroplethSteps[i];
+      if (val < r[0]) {
+        return r[1];
+      } else {
+        i++;
+      }
+    }
+    return "#fff";
+  }
```
Comment (kevinearldenny) [inline src/client/components/evaluate/detail/Compactness.tsx:64]: Is there a more typescript-y way to do this?

File: src/client/components/map/Map.tsx [hunk 2]
```diff
@@ -191,6 +228,36 @@ const DistrictsMap = ({
       );
   };
 
+  // Handle evaluate mode map views
+  useEffect(() => {
+    if (map) {
+      if (evaluateMetric) {
+        // Make the county layer the only layer visible
```
Comment (kevinearldenny) [inline src/client/components/map/Map.tsx:235]:
```suggestion
        // Remove all geolayers from view
```

File: src/client/components/map/index.ts [hunk 3]
```diff
@@ -47,6 +49,32 @@ const filteredLabelLayers = [
   "poi-label"
 ];
 
+export function getChloroplethSteps(metricKey: string) {
+  const compactnessSteps = [
+    [0.3, "#edf8fb"],
+    [0.4, "#b2e2e2"],
+    [0.5, "#66c2a4"],
+    [0.6, "#2ca25f"],
+    [0.6, "#006d2c"]
```
Comment (pcaisse) [inline src/client/components/map/index.ts:58]: I think the last stop should be up to 1 like so (without this my 78% compactness district wasn't showing up as dark green):
```suggestion
    [1.0, "#006d2c"]
```

File: src/client/components/evaluate/detail/Compactness.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  return (
+    <Box sx={{ ml: "20px", overflowY: "scroll" }}>
+      <Heading as="h4" sx={{ variant: "text.h4", display: "block", width: "100%" }}>
+        Average compactness of
+        {Number(metric.value) ? ` ${Math.floor(Number(metric.value) * 100)}%` : " "}
```
Comment (pcaisse) [inline src/client/components/evaluate/detail/Compactness.tsx:70]: See previous comment about casting the potentially undefined `metric.value` to a number

**Review by pcaisse — CHANGES_REQUESTED** [review]
I found a couple of issues with functionality and then added a bunch of comments with suggestions mostly related to type safety.

File: src/client/components/map/Map.tsx [hunk 5]
```diff
@@ -191,6 +230,33 @@ const DistrictsMap = ({
...
+          map.setLayoutProperty(DISTRICTS_COMPACTNESS_CHOROPLETH_LAYER_ID, "visibility", "visible");
+        }
+      } else {
+        // Reset map state to default
+        map.setLayoutProperty(DISTRICTS_LAYER_ID, "visibility", "visible");
+        map.setLayoutProperty(DISTRICTS_COMPACTNESS_CHOROPLETH_LAYER_ID, "visibility", "none");
+        staticMetadata.geoLevelHierarchy.forEach((geoLevel, idx) => {
+          if (idx === staticMetadata.geoLevelHierarchy.length - 1) {
+            map.setLayoutProperty(levelToLineLayerId(geoLevel.id), "visibility", "visible");
+            map.setLayoutProperty(levelToSelectionLayerId(geoLevel.id), "visibility", "visible");
```
Comment (kevinearldenny) [inline src/client/components/map/Map.tsx:253]: @pcaisse The bug encountered with the map not working after switching back to regular Project mode from the Project Evaluate view was due to previously having only added the line layer and not the selection layer

File: src/client/components/evaluate/detail/Compactness.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    display: "inline-block",
+    verticalAlign: "center"
+  },
+  tableHeader: {
+    float: "left",
+    width: "40%"
+  }
+};
+
+const CompactnessMetricDetail = ({
```
[the remark this answers is not in the document]
Comment (kevinearldenny) [reply src/client/components/evaluate/detail/Compactness.tsx:42]: Should be fixed now - was unrelated to this table structure error (which is also fixed)

**Review by pcaisse — APPROVED** [review]
LGTM! :tada:

**Pull request merged**